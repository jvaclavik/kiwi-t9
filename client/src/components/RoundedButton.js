// @flow
import React from 'react'
import { StyleSheet, Text, TouchableOpacity } from 'react-native'

// theme
import { Colors, Metrics } from '../themes'

const styles = StyleSheet.create({
  button: {
    alignItems: 'center',
    backgroundColor: Colors.background,
    borderColor: Colors.primary,
    borderWidth: 2,
    width: Metrics.buttons.height,
    height: Metrics.buttons.height,
    borderRadius: Metrics.buttons.height / 2,
    justifyContent: 'center',
  },
})

type Props = {|
  +onPress: () => void,
  +children: string,
|}

export default class RoundedButton extends React.PureComponent<Props> {
  render() {
    const { children, onPress } = this.props
    return (
      <TouchableOpacity onPress={onPress} style={styles.button}>
        <Text>{children}</Text>
      </TouchableOpacity>
    )
  }
}
