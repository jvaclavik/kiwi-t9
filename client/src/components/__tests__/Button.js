import React from 'react'
import renderer from 'react-test-renderer'

// components
import RoundedButton from '../RoundedButton'

test('renders correctly', () => {
  const tree = renderer
    .create(<RoundedButton onPress={() => null}>Ahoj</RoundedButton>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
