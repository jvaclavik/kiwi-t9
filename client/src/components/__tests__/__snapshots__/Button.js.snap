// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderColor": "rgb(24,202,167)",
      "borderRadius": 42,
      "borderWidth": 2,
      "height": 84,
      "justifyContent": "center",
      "opacity": 1,
      "width": 84,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Ahoj
  </Text>
</View>
`;
